src/enc/backward_references/hq.rs

```diff
@@ -1082,39 +1082,29 @@ pub fn BrotliCreateZopfliBackwardReferences<
 
 fn SetCost(histogram: &[u32], histogram_size: usize, literal_histogram: i32, cost: &mut [floatX]) {
     let mut sum: u64 = 0;
-    let mut missing_symbol_sum: u64;
 
-    let mut i: usize;
-    for i in 0usize..histogram_size {
+    for i in 0..histogram_size {
         sum = sum.wrapping_add(u64::from(histogram[i]));
     }
-    let log2sum: floatX = FastLog2(sum) as (floatX);
-    missing_symbol_sum = sum;
+    let log2sum = FastLog2(sum);
     if literal_histogram == 0 {
-        for i in 0usize..histogram_size {
-            if histogram[i] == 0u32 {
-                missing_symbol_sum = missing_symbol_sum.wrapping_add(1);
+        for i in 0..histogram_size {
+            if histogram[i] == 0 {
+                sum = sum.wrapping_add(1);
```

Comment:
can you keep the variable named "missing_symbol_sum"...it's confusing to a reader why we might compute the log2sum of a "partial" sum... if it's called as a different variable then the logic makes sense.

if you want to make it clearer you could reduce the scope of "sum" above to a nested block and then have missing_symbol_sum be defined outside the block (and assigned in it)

Reply:
thx, fixed